pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -940,3 +977,164 @@ func (r *ReplicatedUpgradeReconciler) getSandboxPodFacts(ctx context.Context, do
 	}
 	return r.PFacts[r.sandboxName], nil
 }
+
+// removeReplicaGroupAFromVdb will remove subclusters of replica group A from VerticaDB
+func (r *ReplicatedUpgradeReconciler) removeReplicaGroupAFromVdb(ctx context.Context) (ctrl.Result, error) {
+	// if the sandbox is still there, we wait for promote_sandbox to be done
+	if r.VDB.GetSandboxStatus(r.sandboxName) != nil {
+		return ctrl.Result{Requeue: true}, nil
+	}
+	// if replica group A doesn't contain any subclustesr,
+	// we skip removing the old main cluster
+	if r.countSubclustersForReplicaGroup(vmeta.ReplicaGroupAValue) == 0 {
+		return ctrl.Result{}, nil
+	}
+
+	r.Log.Info("starting removal of replica group A from VerticaDB")
+
+	scNames := r.getSubclustersForReplicaGroup(vmeta.ReplicaGroupAValue)
+	scNameSetForGroupA := make(map[string]any)
+	for _, sc := range scNames {
+		scNameSetForGroupA[sc] = struct{}{}
+	}
+	updateSubclustersInVdb := func() (bool, error) {
+		// remove subclusters in replica group A
+		removed := false
+		for i := len(r.VDB.Spec.Subclusters) - 1; i >= 0; i-- {
+			_, found := scNameSetForGroupA[r.VDB.Spec.Subclusters[i].Name]
+			if found {
+				r.VDB.Spec.Subclusters = append(r.VDB.Spec.Subclusters[:i], r.VDB.Spec.Subclusters[i+1:]...)
+				removed = true
+			}
+		}
+		return removed, nil
+	}
+
+	updated, err := vk8s.UpdateVDBWithRetry(ctx, r.VRec, r.VDB, updateSubclustersInVdb)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to delete subclusters in old main cluster: %w", err)
+	}
+	if updated {
+		r.Log.Info("deleted subclusters in old main cluster", "subclusters", scNames)
```

Comment:
I have not deleted the old main yet. The nodes are still up and scs still exist.